### PR TITLE
Improve Readmes and code commenting

### PR DESCRIPTION
**Pull request by benitav**
* Leaner Readme's with details moved to our docs SDK references. 
* More consistent Readme's across RN and Web (Flutter is TODO separately).
* Added JSDoc comments for common methods and properties

- [x] This PR is dependent on merging this Gitbook change request: https://app.gitbook.com/o/YjXqDRfjWRhJMPvQn7Cm/s/V9geAO9ITPi8WOYK5o0r/~/diff/~/changes/391/client-sdk-references/react-native-and-expo

### CODE REVIEW

File: packages/powersync-sdk-common/src/client/sync/bucket/OpType.ts [hunk 1]
```diff
@@ -7,6 +7,9 @@ export enum OpTypeEnum {
 
 export type OpTypeJSON = string;
 
+/**
```
Comment (DominicGBauer) [inline packages/powersync-sdk-common/src/client/sync/bucket/OpType.ts:10]: if this is internal should we not ignore it?

File: packages/powersync-sdk-common/src/db/DBAdapter.ts [hunk 2]
```diff
@@ -42,12 +42,14 @@ export interface DBGetUtils {
 }
 
 export interface LockContext extends DBGetUtils {
-  /** Execute a statement and optionally return results. */
+  /** Execute a single write statement. */
```
Comment (DominicGBauer) [inline packages/powersync-sdk-common/src/db/DBAdapter.ts:45]: are you sure this is limited to `writes` ?

File: packages/powersync-sdk-common/src/utils/BaseObserver.ts [hunk 3]
```diff
@@ -15,6 +15,9 @@ export class BaseObserver<T extends BaseListener = BaseListener> implements Base
     this.listeners = {};
   }
 
+  /**
+   * Register a listener for updates on the PowerSync client.
```
Comment (DominicGBauer) [inline packages/powersync-sdk-common/src/utils/BaseObserver.ts:19]: would it not be updates to* as oposed to on?

File: packages/powersync-sdk-common/src/db/DBAdapter.ts (hunk 2 above)
Comment (benitav) [reply packages/powersync-sdk-common/src/db/DBAdapter.ts:45]: mm, according to https://pub.dev/documentation/powersync/latest/powersync/PowerSyncDatabase/execute.html and https://docs.powersync.com/client-sdk-references/react-native-and-expo#using-powersync I thought so yes, but perhaps I mapped it incorrectly here?

File: packages/powersync-sdk-common/src/client/sync/bucket/OpType.ts (hunk 1 above)
Comment (benitav) [reply packages/powersync-sdk-common/src/client/sync/bucket/OpType.ts:10]: Possibly; I'm keeping ignores for a future PR.

File: packages/powersync-sdk-common/src/db/DBAdapter.ts (hunk 2 above)
Comment (DominicGBauer) [reply packages/powersync-sdk-common/src/db/DBAdapter.ts:45]: Yeah I think you are correct. Thanks

**Pull request merged**